test/e2e/security_test.go

```diff
@@ -121,7 +129,14 @@ func testUserProvidedCert(t *testing.T, data *TestData) {
 
 // testSelfSignedCert tests the selfSignedCert=true case.
 func testSelfSignedCert(t *testing.T, data *TestData) {
-	testCert(t, data, "", true)
+	secretBeforeRestart, err := data.clientset.CoreV1().Secrets(tlsSecretNamespace).Get(context.TODO(), tlsSecretName, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	testCert(t, data, string(secretBeforeRestart.Data[certificate.TLSCertFile]), true)
```

Comment:
so in the self-signed certificate case, the published CA bundle (published to the ConfigMap) is equal to the certificate itself (stored in the Secret)?